Sources/MapboxMaps/Location/LocationStyleProtocol.swift

```diff
@@ -8,15 +8,30 @@ internal protocol LocationStyleProtocol: AnyObject {
     func setSourceProperty(for sourceId: String, property: String, value: Any) throws
 
     //swiftlint:disable function_parameter_count
-    func addImage(_ image: UIImage, id: String, sdf: Bool, stretchX: [ImageStretches], stretchY: [ImageStretches], content: ImageContent?) throws
+    func addImage(_ image: UIImage,
+                  id: String,
+                  sdf: Bool,
+                  stretchX: [ImageStretches],
+                  stretchY: [ImageStretches],
+                  content: ImageContent?) throws
 }
 
 extension LocationStyleProtocol {
     internal func addPersistentLayer(_ layer: Layer, layerPosition: LayerPosition? = nil) throws {
         try addPersistentLayer(layer, layerPosition: layerPosition)
     }
 
-    internal func addImage(_ image: UIImage, id: String, sdf: Bool = false, stretchX: [ImageStretches] = [], stretchY: [ImageStretches] = [], content: ImageContent? = nil) throws {
-        try addImage(image, id: id, sdf: sdf, stretchX: stretchX, stretchY: stretchY, content: content)
+    internal func addImage(_ image: UIImage,
+                           id: String,
+                           sdf: Bool,
+                           stretchX: [ImageStretches],
+                           stretchY: [ImageStretches],
+                           content: ImageContent?) throws {
+        try addImage(image,
+                     id: id,
+                     sdf: sdf,
+                     stretchX: stretchX,
+                     stretchY: stretchY,
+                     content: content)
```

Comment:
Trying to understand how this isn't a recursive call

Reply:
This is in a protocol extension which must have something to do with it. Let's see if we can find a way not to do this.

Reply:
I have to dig into this further, I'm not sure how involved the `LocationStyleProtocol` is.

Reply:
I tried this locally — if you just delete this entire protocol extension, there just end up being 2 places that were depending on the default parameter for `layerPosition` assigned here where you'll need to explicitly pass nil instead. I think that's a pretty reasonable tradeoff for deleting these 20 lines of confusing code.